scripts/jira_issue_flow.py

```diff
@@ -0,0 +1,173 @@
+        description (str): The description of the issue.
+
+    Returns:
+        Issue: The existing or newly created Jira issue.
+
+    Raises:
+        KeyError: If required environment variables are missing.
+    """
+    if issue := get_jira_issue_by_jql(jql=jql, summary=summary):
+        return issue
+
+    return create_jira_issue(summary=summary, issue_type=issue_type, parent_key=parent_key, description=description)
+
+
+def get_or_create_story(summary: str, epic_key: str, description: str = "") -> Issue:
+    """
+    Retrieves or creates a Jira Story linked to the specified Epic.
+
+    Args:
+        summary (str): The summary of the Story.
+        epic_key (str): The key of the Epic.
+        description (str): The description of the Story.
+
+    Returns:
+        Issue: The existing or newly created Story issue.
+    """
+    project = os.environ["MIGRATION_BACKLOG_JIRA_PROJECT"]
+    jql = f'project = {project} AND summary ~ "{summary}" AND issuetype = Story AND "Epic Link" = "{epic_key}"'
+
+    return get_or_create_jira_issue(
+        summary=summary,
+        jql=jql,
+        issue_type="Story",
+        parent_key=epic_key,
+        description=description,
+    )
+
+
+def get_or_create_subtask(summary: str, story_key: str, description: str = "") -> Issue:
+    """
+    Retrieves or creates a Jira Sub-task under the specified Story.
+
+    Args:
+        summary (str): The summary of the Sub-task.
+        story_key (str): The key of the parent Story.
+        description (str): The description of the Sub-task.
+
+    Returns:
+        Issue: The existing or newly created Sub-task issue.
+    """
+    project = os.environ["MIGRATION_BACKLOG_JIRA_PROJECT"]
+    jql = f'project = {project} AND summary ~ "{summary}" AND issuetype = Sub-task AND parent = "{story_key}"'
+
+    return get_or_create_jira_issue(
+        summary=summary,
+        jql=jql,
+        issue_type="Sub-task",
+        parent_key=story_key,
+        description=description,
+    )
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Create Jira Story and Sub-task if not present.")
+    parser.add_argument("--story-summary", required=True, help="Summary for the story.")
+    parser.add_argument("--story-description", default="", help="Description for the story.")
+    parser.add_argument("--subtask-summary", required=True, help="Summary for the sub-task.")
+    parser.add_argument("--subtask-description", default="", help="Description for the sub-task.")
+    args = parser.parse_args()
+
+    epic_key = os.environ["MIGRATION_BACKLOG_JIRA_EPIC_KEY"]
+
+    story = get_or_create_story(
+        summary=args.story_summary,
+        epic_key=epic_key,
+        description=args.story_description,
+    )
+    get_or_create_subtask(
+        summary=args.subtask_summary,
+        story_key=story.key,
+        description=args.subtask_description,
+    )
```

Comment:
I would add some logging indicating the success of the story or subtask created...

Reply:
done 
![image](https://github.com/user-attachments/assets/b81b5a84-d91b-48b9-a628-fd92f4535761)